lib/graphql_schema.rb

```diff
@@ -200,6 +200,11 @@ def fields(include_deprecated: false)
       include_deprecated ? @fields : @fields.reject(&:deprecated?)
     end
 
+    def field_from_name(field_name, include_deprecated: false)
+      @fields_by_name ||= fields(include_deprecated: include_deprecated).map{ |field| [field.name, field]}.to_h
```

Comment:
Maybe this should just be.
```
def fields_by_name
    @fields_by_name ||= fields(include_deprecated: true).map{ |field| [field.name, field]}.to_h
end
```
And then the user can check later if the field is deprecated.